chain/chain/src/chain.rs

```diff
@@ -1958,6 +1958,10 @@ impl Chain {
                 preprocess_res
             }
             Err(e) => {
+                if e.is_bad_data() {
+                    metrics::NUM_INVALID_BLOCKS.inc();
+                    self.invalid_blocks.put(*block.hash(), ());
```

Comment:
in separate PR:

I wonder if we could export the counter for each error type (and similarly - if we could export/show the error type in the debug page)

Reply:
Good idea!